### PR TITLE
Build: Add config files for jscs and jshint, run jshint with npm-test

### PR DESCRIPTION
**Pull request by jzaefferer**
Fixes 3 'lint' issues: Remove unused var, replace semicolon with comma, define Promise (even though Promise is 'now' native in nodejs).

There's currently lots of whitespace issues, those need to be addressed separately, then jscs can be regularly checked as well.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
   ],
   "scripts": {
     "start": "./bin/server.js",
-    "test": "nodeunit"
+    "test": "jshint bin/ lib/ test/ *.js && nodeunit"
```
Comment (arschmitz) [inline package.json]: Shouldnt we add jscs here as well?

Comment (jzaefferer) [inline package.json]: I mentioned this in the commit message (copied into the PR description):

> There's currently lots of whitespace issues, those need to be addressed separately, then jscs can be regularly checked as well.

Comment (jzaefferer) [inline package.json]: Though I could add `"requireSpacesInsideParentheses": null` to the config to turn of that one rule and fix the rest.

Comment (arthurvr) [inline package.json]: Should the `exceptions/` directory also be linted? I know it's just some arrays but there are still simple mistakes we can make there.

Comment (jzaefferer) [inline package.json]: I'll add that, there's just two issues that I can fix quickly enough.

Comment (arschmitz) [inline package.json]: if its just white space just run jscs with fix and do it as part of this its a small enough repo.

**Comment by jzaefferer** [comment]
Updated, now running both jshint and jscs as part of npm-test.

**Comment by jzaefferer** [comment]
Updated again, now jshint and jscs validate all js files in the repo (both tools ignore node_modules, it seems), whitespace issues are fixed as well.

**Comment by arschmitz** [comment]
Looks good to me. :+1:

**Pull request merged**